source/Ship.cpp

```diff
@@ -1244,15 +1244,27 @@ bool Ship::Move(list<Effect> &effects, list<shared_ptr<Flotsam>> &flotsam)
 		}
 
 
-        double oocShieldRate = attributes.Get("ooc shield generation");
-        if( ! IsInCombat() && oocShieldRate > 0.)
-        {
+		double oocShieldRate = attributes.Get("ooc shield generation");
+		if( ! IsInCombat() && oocShieldRate > 0.)
+		{
 			double shieldEnergy = attributes.Get("ooc shield energy");
 			double shieldHeat = attributes.Get("ooc shield heat");
 			double shieldsAdded = AddShieldsOOC(oocShieldRate * min(1., shieldEnergy ? energy / shieldEnergy : 1.));
 			energy -= shieldEnergy * shieldsAdded / oocShieldRate;
 			heat += shieldHeat * shieldsAdded / oocShieldRate;
-        }
+		}
+
+		double overShieldRate = attributes.Get("overclocked shield generation");
+		if( overShieldRate > 0.)
+		{
+			double shieldEnergy = attributes.Get("overclocked shield energy");
+			double shieldHeat = attributes.Get("overclocked shield heat");
+			double shieldsAdded = AddShieldsOverclocked(overShieldRate * min(1., shieldEnergy ? energy / shieldEnergy : 1.));
```

Comment:
Is it really a good idea to have shield generation be dependent on the level of energy drain?